ui/appui-react/src/test/statusfields/SnapMode.test.tsx

```diff
@@ -69,12 +69,62 @@ describe("SnapModeField", () => {
         <SnapModeField />
       </Provider>
     );
+    const previousMode = UiFramework.getAccudrawSnapMode();
     await theUserTo.click(screen.getByRole("button"));
     await theUserTo.click(screen.getByText("snapModeField.bisector"));
 
-    expect(UiFramework.getAccudrawSnapMode()).to.equal(SnapMode.Bisector);
+    expect(UiFramework.getAccudrawSnapMode()).to.equal(
+      previousMode | SnapMode.Bisector
+    );
     expect(spy.mock.calls[0][0].eventIds.values()).toContain(
       "configurableui:set_snapmode"
     );
   });
+
+  it("should not be able to disable all snapMode. at least one should be active at all time", async () => {
+    render(
+      <Provider store={TestUtils.store}>
+        <SnapModeField />
+      </Provider>
+    );
+    const previousMode = UiFramework.getAccudrawSnapMode();
+    expect(previousMode).toEqual(SnapMode.NearestKeypoint); // Make sure that only one mode is active.
+    await theUserTo.click(screen.getByRole("button"));
+    await theUserTo.click(screen.getByText("snapModeField.keypoint")); // Click to deactivate the mode.
+    expect(UiFramework.getAccudrawSnapMode()).to.equal(
+      SnapMode.NearestKeypoint
+    ); // Expect it to be still active.
+  });
+
+  it("Status Bar only display the chosen SnapModes.", () => {
+    const { container } = render(
+      <Provider store={TestUtils.store}>
+        <SnapModeField
+          snapModes={
+            // Filter out the Bisector snap mode.
+            Object.values(SnapMode).filter(
+              (snapMode) => snapMode !== SnapMode.Bisector
+            ) as SnapMode[]
+          }
+        />
+      </Provider>
+    );
+
+    const button = container.querySelector("button");
+    expect(button).toBeTruthy();
+    fireEvent.click(button!);
+
+    // Bisector snap mode should not be present.
+    expect(screen.queryByText("snapModeField.bisector")).toBeFalsy();
+
+    const iconContainer = container.querySelector(".icon");
+    expect(iconContainer).toBeTruthy();
+
+    const snaps = container.parentElement!.querySelectorAll(
+      ".nz-footer-snapMode-snap"
+    );
+    expect(snaps.length).to.eql(6); // 7 modes total, minus Bisector = 6 modes
```

Comment:
Avoid styling/markup assertions, instead prefer byText assertions.

Reply:
When writing the tests, Bill used the `container` instead of using the `screen` variable. 
Like this, the test can be synchronous, but we cant get elements by text. 
I wonder if the performance gain from writing synchronous code is something we should consider, or if we should just go with the most robust selectors (getByText / Id) ?

Reply:
Just go with getByText, which is best option and is available in this case